### ISSUE TITLE
Track reactions with the effect

### ISSUE THREAD

**Issue by NdranC** [reyzor1991/foundry-vtt-pf2e-reaction#42]
I'm not 100% sure if this is intended or a bug, but after doing some testing I'm noticing that the module seems to have an internal tracker for reaction that has nothing to do with the "Effect: Reaction was used" effect. 
Let's say I trigger an Attack of Opportunity from a creature. I click yes on the chat message which adds the "Effect: Reaction was used" to the creature. But as a GM I regret my decision and don't want to use it after all. If I remove the effect from the creature I would expect the reaction to come back and to be usable again but it doesn't. It will the npc starts their turn.
Similar to that, if I use a reaction that the mod doesn't track, I would love to just add the effect "Effect: Reaction was used" to stop bothering me about other reactions.

Basically, I would like the mod to actually use the effect to keep track of reactions instead of just working as a decoration.
[Screencast from 2024-09-04 16-56-55.webm](https://github.com/user-attachments/assets/23b48baa-3c36-4600-a0b1-72be1b836d2e)
This is an example of my first point.

**Comment by reyzor1991** [comment]
It's not a bug because effect is only visual and represent that all reactions were used

For such thing hard to track logic as revert and etc

**Comment by NdranC** [comment]
I see. What a shame. I'm used to do it that way from midiqol in the dnd side of things.

**Comment by reyzor1991** [comment]
I could try to write smth for rollback but it's not a fast
